src/main/java/com/danielfrak/code/keycloak/providers/rest/LegacyProvider.java

```diff
@@ -89,4 +92,23 @@ public boolean isConfiguredFor(RealmModel realmModel, UserModel userModel, Strin
     public void close() {
         // Not needed
     }
+
+    @Override
+    public boolean updateCredential(RealmModel realm, UserModel user, CredentialInput input) {
+        if (!user.getFederationLink().isBlank()) {
```

Comment:
Sorry for not noticing this earlier, but I'm fairly sure that this will result in a `NullPointerException` if `user.getFederationLink()` is null. This should do it, though:

```
if (!(user.getFederationLink() == null || user.getFederationLink().isBlank()) {
  // ...
}
```